k9mail-library/src/main/java/com/fsck/k9/mail/autoconfiguration/DNSHelper.java

```diff
@@ -0,0 +1,22 @@
+package com.fsck.k9.mail.autoconfiguration;
+
+
+import java.net.UnknownHostException;
+
+import org.xbill.DNS.MXRecord;
+import org.xbill.DNS.TextParseException;
+
+
+public class DNSHelper {
+    public static String getMXDomain(String domain) throws TextParseException, UnknownHostException {
+        DNSOperation dnsOperation = new DNSOperation();
+        MXRecord mxRecord = dnsOperation.mxLookup(domain);
+        if (mxRecord != null) {
+            final String target = mxRecord.getTarget().toString(true);
+            final String[] targetParts = target.split("\\.");
```

Comment:
it's not immediately obvious what this does, please extract it into a method (e.g. `getDomainFromFqdn`)